frontend_app_gallery/src/container/Gallery.jsx

```diff
@@ -154,22 +166,48 @@ class Gallery extends React.Component {
     // app crash telling it cannot render a Link outside a router
     // see https://github.com/tracim/tracim/issues/1637
     // GLOBAL_dispatchEvent({type: 'setBreadcrumbs', data: {breadcrumbs: breadcrumbsList}})
-    this.setState({ breadcrumbsList: breadcrumbsList })
+    this.setState({ breadcrumbsList, breadcrumbsLoaded: !includeFile })
   }
 
-  loadFolderDetail = async (workspaceId, folderId) => {
+  loadFolderDetailAndParentsDetails = async (workspaceId, folderId) => {
     const { state, props } = this
 
-    const fetchContentDetail = await handleFetchResult(
+    let folderDetail
+
+    let fetchContentDetail = await handleFetchResult(
       await getFolderDetail(state.config.apiUrl, workspaceId, folderId)
     )
 
     switch (fetchContentDetail.apiResponse.status) {
       case 200:
-        this.setState({ fileName: fetchContentDetail.body.filename })
+        folderDetail = {}
```

Comment:
don't initialize your variable to undefined and then reinitialize it to an empty object. Just initialize it to what you need. It'll also fix the fact that your function can return undefined